doc/user/layouts/partials/create-source/envelope/debezium/syntax.html

```diff
@@ -1 +1,2 @@
 **ENVELOPE DEBEZIUM** | Use the Debezium envelope, which uses a diff envelope to handle CRUD operations. This option requires payloads have the [appropriate fields](#format-implications), and is generally only supported by sources published to Kafka by [Debezium]. For more information, see [Debezium envelope details](#debezium-envelope-details).
+**ENVELOPE DEBEZIUM UPSERT** | An extension of **ENVELOPE DEBEZIUM**, Materialize maintains enough state to handle log-compaction on Kafka topics carrying Debezium data.
```

Comment:
Can this be simplified into:

> Use the Debezium envelope with `UPSERT` enabled. This is strongly recommended for Kafka sources with log compaction, but increases memory consumption.

I would like to say `required` here, since it sounds like user will get unpredictable/invalid results with `ENVELOPE DEBEZIUM`, but it also sounds like we don't enforce anything. Maybe we should add 

> This option is incompatible with Kafka compaction

to `ENVELOPE DEBEZIUM`.